<a href="https://colab.research.google.com/github/kay-kay-t/CodingDojo/blob/main/Project2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [18]:
# Imports (for now)
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

## Dataset 1: Video content prediction

In [19]:
# Load data 
df1 = pd.read_csv('/content/drive/MyDrive/Data/public_meta_df.csv')
df1.head()

comp_id  ad_blocked  embed    ratio  ...  n_tags  n_formats  dayofweek  hour
0     3001       False   True  1.33333  ...      10          2          2     3
1     3002       False   True  1.33333  ...      10          2          2     3
2     3003       False   True  1.33333  ...      10          2          2     3
3     3004       False   True  1.77778  ...       6          5          2     3
4     3005       False   True  1.33333  ...       6          2          2     3

[5 rows x 13 columns]

In [20]:
df1['n_likes'].value_counts()

0    921
1     47
2     12
3      5
6      1
Name: n_likes, dtype: int64

In [21]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 986 entries, 0 to 985
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   comp_id         986 non-null    int64  
 1   ad_blocked      986 non-null    bool   
 2   embed           986 non-null    bool   
 3   ratio           986 non-null    float64
 4   duration        986 non-null    int64  
 5   language        986 non-null    int64  
 6   partner         986 non-null    bool   
 7   partner_active  986 non-null    bool   
 8   n_likes         986 non-null    int64  
 9   n_tags          986 non-null    int64  
 10  n_formats       986 non-null    int64  
 11  dayofweek       986 non-null    int64  
 12  hour            986 non-null    int64  
dtypes: bool(4), float64(1), int64(8)
memory usage: 73.3 KB


In [22]:
df1.shape

(986, 13)

First choice: dataset 1

1) Source of data
[Kaggle](https://www.kaggle.com/prokaggler/video-popularity-prediction)

2) Brief description of data

One of datasets from Kaggle, that has different features of video. I can make some predictions on different features based on other features video has. For exaple, I can make  prediction on how many likes video can get or how likely it will have a partner, 

3) What is the target?

Number of likes / Partner

4) Is this a classification or regression problem?

Regression problem / classification problem

5) How many features?

11 features

6) How many rows of data.

985

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

I will probably need to use ohe, can try different regression models

## Dataset 2: New York City Airbnb Open Data

In [23]:
df2 = pd.read_csv('/content/drive/MyDrive/Data/AB_NYC_2019.csv')
df2.head()

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0

[5 rows x 16 columns]

In [24]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [25]:
df2.shape

(48895, 16)

Second choice: dataset 2

1) Source of data
[Kaggle](https://www.kaggle.com/dgomonov/new-york-city-airbnb-open-data?select=AB_NYC_2019.csv)

2) Brief description of data

This dataset describes the listing activity and metrics in NYC, NY for 2019. I can make different kinds of predictions, like price or number of reviews, or roomtype type let's say.

3) What is the target?

Price / number of reviews / room type

4) Is this a classification or regression problem?

Regression problem / classification problem

5) How many features?

11

6) How many rows of data.

48894 

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

Looks like this data has some missing values, ohe, different models